vformer/models/classification/visformer.py

```diff
@@ -0,0 +1,351 @@
+        x = self.drop(x)
+        x = x + xt
+        del xt
+
+        return x
+
+
+@MODEL_REGISTRY.register()
+class Visformer_Attention_Block(nn.Module):
+    """
+    Attention Block for Vision-Friendly transformers
+    https://arxiv.org/abs/2104.12533
+
+    Parameters
+    ----------
+    in_channels: int
+        Number of input channels
+    num_heads: int
+        Number of heads for attention, default is 8
+    activation: torch.nn.Module
+        Activation function between layers, default is nn.GELU
+    p_dropout: float
+        Dropout rate, default is 0.0
+    """
+
+    def __init__(self, in_channels, num_heads=8, activation=nn.GELU, p_dropout=0.0):
+        super().__init__()
+
+        self.conv1 = nn.Conv2d(in_channels, in_channels * 4, kernel_size=1, bias=False)
+        self.act1 = activation()
+        self.conv2 = nn.Conv2d(in_channels * 4, in_channels, kernel_size=1, bias=False)
+        self.attn = VanillaSelfAttention(
+            in_channels, num_heads=num_heads, head_dim=in_channels // num_heads
+        )
+        self.norm1 = nn.BatchNorm2d(in_channels)
+        self.norm2 = nn.BatchNorm2d(in_channels)
+        self.drop = nn.Dropout(p_dropout)
+
+    def forward(self, x):
+
+        B, C, H, W = x.shape
+        xt = einops.rearrange(x, "b c h w -> b (h w) c")
+        x = self.norm1(x)
+        x = einops.rearrange(x, "b c h w -> b (h w) c")
+        x = self.attn(x)
+        x = xt + x
+        x = einops.rearrange(x, "b (h w) c -> b c h w", h=H, w=W)
+        xt = x
+        x = self.norm2(x)
+        x = self.conv1(x)
+        x = self.act1(x)
+        x = self.conv2(x)
+        x = self.drop(x)
+        x = xt + x
+        del xt
+
+        return x
+
+
+@MODEL_REGISTRY.register()
+class Visformer(nn.Module):
+    """
+    A builder to construct a Vision-Friendly transformer model as in the paper: "Visformer: A vision-friendly transformer"
+    https://arxiv.org/abs/2104.12533
+
+    Parameters
+    ----------
+    img_size: int,tuple
+        Size of the input image
+    num_classes: int
+        Number of classes in the dataset
+    depth: tuple[int]
+        Number of layers before each embedding reduction
+    config: tuple[int]
+        Choice of convolution block (0) or attention block (1) for corresponding layer
+    channel_config: tuple[int]
+        Number of channels for each layer
+    num_heads: int
+        Number of heads for attention block, default is 8
+    conv_group: int
+        Number of groups for convolution block, default is 8
+    p_dropout_conv: float
+        Dropout rate for convolution block, default is 0.0
+    p_dropout_attn: float
+        Dropout rate for attention block, default is 0.0
+    activation: torch.nn.Module
+        Activation function between layers, default is nn.GELU
+    pos_embedding: bool
+        Whether to use positional embedding, default is True
+
+    """
+
+    def __init__(
+        self,
+        img_size,
+        n_classes,
+        depth: tuple,
```

Comment:
```suggestion
    def __init__(
        self,
        img_size,
        num_classes,
        depth: tuple,
```
As per the docstring

Reply:
changes the docstring to maintain common naming in package